ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -107,15 +101,16 @@ McapHandler::~McapHandler()
 }
 
 void McapHandler::add_schema(
-        const fastrtps::types::DynamicType_ptr& dynamic_type)
+        const fastdds::dds::DynamicType::_ref_type& dynamic_type,
+        const fastdds::dds::xtypes::TypeIdentifier& type_identifier)
```

Comment:
Assert that this type_identifier is EK_COMPLETE. I would add this assertion before calling this method in TypeObjectWriter.